src/app/Plans/components/Wizard/SelectVMs.tsx

```diff
@@ -0,0 +1,141 @@
+import * as React from 'react';
+import { Pagination } from '@patternfly/react-core';
+import {
+  Table,
+  TableHeader,
+  TableBody,
+  sortable,
+  compoundExpand,
+  classNames as classNamesTransform,
+  ICell,
+  IRow,
+  wrappable,
+} from '@patternfly/react-table';
+import tableStyles from '@patternfly/react-styles/css/components/Table/table';
+
+import { IVMs } from '../../types';
+import { useSortState, usePaginationState, useSelectionState } from '@app/common/hooks';
+import { StatusIcon, StatusType } from '@konveyor/lib-ui';
+
+interface ISelectVMsProps {
+  VMs: IVMs[];
+}
+
+const SelectVMs: React.FunctionComponent<ISelectVMsProps> = ({ VMs }: ISelectVMsProps) => {
+  const getSortValues = (vm: IVMs) => {
+    return ['', vm.Name, vm.Datacenter, vm.Cluster, vm.Host, vm.FolderPath, ''];
+  };
+
+  const { sortBy, onSort, sortedItems } = useSortState(VMs, getSortValues);
+  const { currentPageItems, setPageNumber, paginationProps } = usePaginationState(sortedItems, 10);
+  React.useEffect(() => setPageNumber(1), [sortBy, setPageNumber]);
+
+  const { selectedItems, toggleItemSelected, areAllSelected, selectAll } = useSelectionState<IVMs>(
+    sortedItems
+  );
+
+  const { selectedItems: expandedVMs, toggleItemSelected: toggleVMExpanded } = useSelectionState<
+    IVMs
+  >(sortedItems);
+
+  const columns: ICell[] = [
+    {
+      // Using a custom column instead of Table's onSelect prop due to issues
+      title: (
+        <input
+          type="checkbox"
+          aria-label="Select all VMs"
+          onChange={(event: React.FormEvent<HTMLInputElement>) => {
+            selectAll(event.currentTarget.checked);
+          }}
+          checked={areAllSelected}
+        />
+      ),
+      columnTransforms: [classNamesTransform(tableStyles.tableCheck)],
+    },
+    {
+      title: 'Migration Analysis',
+      transforms: [sortable, wrappable],
+      cellTransforms: [compoundExpand],
```

Comment:
We'll also need to remove `compoundExpand` here and remove the import.

Reply:
@mturley are going for a Collapsible Table?

@vconzola, Initially I tried an Expandable Table but that didn't fit the purpose so I thought about a Compound Expandable Table.